homeassistant/components/meteo_france/weather.py

```diff
@@ -1,125 +1,197 @@
 
     @property
     def condition(self):
         """Return the current condition."""
-        return self.format_condition(self._data["weather"])
+        return format_condition(
+            self.coordinator.data.current_forecast["weather"]["desc"]
+        )
 
     @property
     def temperature(self):
         """Return the temperature."""
-        return self._data["temperature"]
-
-    @property
-    def humidity(self):
-        """Return the humidity."""
-        return None
+        return self.coordinator.data.current_forecast["T"]["value"]
 
     @property
     def temperature_unit(self):
         """Return the unit of measurement."""
         return TEMP_CELSIUS
 
+    @property
+    def pressure(self):
+        """Return the pressure."""
+        return self.coordinator.data.current_forecast["sea_level"]
+
+    @property
+    def humidity(self):
+        """Return the humidity."""
+        return self.coordinator.data.current_forecast["humidity"]
+
     @property
     def wind_speed(self):
         """Return the wind speed."""
-        return self._data["wind_speed"]
+        # convert from API m/s to km/h
+        return round(self.coordinator.data.current_forecast["wind"]["speed"] * 3.6)
 
     @property
     def wind_bearing(self):
         """Return the wind bearing."""
-        return self._data["wind_bearing"]
-
-    @property
-    def attribution(self):
-        """Return the attribution."""
-        return ATTRIBUTION
+        wind_bearing = self.coordinator.data.current_forecast["wind"]["direction"]
+        if wind_bearing != -1:
+            return wind_bearing
 
     @property
     def forecast(self):
         """Return the forecast."""
-        reftime = dt_util.utcnow().replace(hour=12, minute=0, second=0, microsecond=0)
-        reftime += timedelta(hours=24)
-        _LOGGER.debug("reftime used for %s forecast: %s", self._data["name"], reftime)
         forecast_data = []
-        for key in self._data["forecast"]:
-            value = self._data["forecast"][key]
-            data_dict = {
-                ATTR_FORECAST_TIME: reftime.isoformat(),
-                ATTR_FORECAST_TEMP: int(value["max_temp"]),
-                ATTR_FORECAST_TEMP_LOW: int(value["min_temp"]),
-                ATTR_FORECAST_CONDITION: self.format_condition(value["weather"]),
-            }
-            reftime = reftime + timedelta(hours=24)
-            forecast_data.append(data_dict)
+
+        if self._mode == FORECAST_MODE_HOURLY:
+            today = time.time()
+            for forecast in self.coordinator.data.forecast:
+                # Can have data in the past
+                if forecast["dt"] < today:
+                    _LOGGER.debug(
+                        "remove forecast in the past: %s %s", self._mode, forecast
+                    )
+                    continue
+                forecast_data.append(
+                    {
+                        ATTR_FORECAST_TIME: self.coordinator.data.timestamp_to_locale_time(
```

Comment:
The forecast datetime should be isoformatted utc time.

https://developers.home-assistant.io/docs/core/entity/weather#forecast